modules/caddyhttp/fileserver/matcher.go

```diff
@@ -440,8 +446,14 @@ func (m MatchFile) selectFile(r *http.Request) (bool, error) {
 
 			candidates := makeCandidates(pattern)
 			for _, c := range candidates {
+				if m.TryPolicy == tryPolicyFirstExistOrFallback && i == maxI {
```

Comment:
I think a comment to explain what the fallback is doing would help (i.e. to skip the IO)